core-api/src/services/preference.service.ts

```diff
@@ -0,0 +1,79 @@
+import axios from 'axios';
+import adapter from 'axios/lib/adapters/http';
+
+/**
+ * @namespace Services
+ * @class PreferenceService
+ */
+export class PreferenceService {
+
+    /**
+     * Creates a new preference in the database
+     * @param userId
+     * @param placeCategoryId
+     * @param checkoutSlotId
+     */
+    public static async create(userId: string, placeCategoryId: string, checkoutSlotId: string): Promise<{}> {
+        try {
+            const preference = await axios.post('http://preference:8081/preferences/create', {
+                userId,
+                placeCategoryId,
+                checkoutSlotId
+            }, { adapter });
+            return preference.data;
+        } catch (e) {
+            console.info(e.message);
+        }
+    }
+
+    /**
+    * Find preference by id
+    * @param id
+    */
+    public static async findById(id: string): Promise<{}> {
+        try {
+            const preference = await axios.get(`http://preference:8081/preferences/${id}`, { adapter });
+            return preference.data;
+        } catch (e) {
+            console.info(e.message);
+        }
+    }
+
+    /**
+    * Find preference by userId
+    * @param userId
+    */
+    public static async findByUserId(userId: string): Promise<{}> {
+        try {
+            const preference = await axios.get(`http://preference:8081/preferences/users/${userId}`, { adapter });
+            return preference.data;
+        } catch (e) {
+            console.info(e.message);
+        }
+    }
+
+    /**
+    * Find preference by placeCategoryId
+    * @param placeCategoryId
+    */
+    public static async findByPlaceCategoryId(placeCategoryId: string): Promise<{}> {
+        try {
+            const preference = await axios.get(`http://preference:8081/preferences/place-categories/${placeCategoryId}`, { adapter });
+            return preference.data;
+        } catch (e) {
+            console.info(e.message);
+        }
+    }
+
+    /**
+     * Find all preferences in the database
+     */
+    public static async findAllPreferences(): Promise<{}> {
+        try {
+            const preferences = await axios.get('http://preference:8081/preferences/', { adapter });
+            return preferences.data;
```

Comment:
I think we should not forget a fallback logic around here. If the preference-service is down for some reason, the core-api should still be able to return something to the client/front-end. 

For example: if we have to implement an endpoint that returns a list of places that a user might like, we need the user's preferences. But if the preference-service returns a `500` status code or simply doesn't respond in a reasonable time, it's not okay for the client to notice it. We could simply make it return a list of places nearby. That wouldn't be the perfect case, but surely would be helpful for a hangry customer xD